### PR TITLE
MEN-4385: acceptance tests for the management end-points to get and set configs

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

File: tests/tests/test_management_config.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            "reported": {},
+            "configured": {
+                "key": "value",
+                "another-key": "another-value",
+                "dollar-key": "$",
+            },
+        } == {k: data[k] for k in ("id", "reported", "configured")}
+        assert "updated_ts" in data.keys()
+        #
+        # replace the configuration
```
Comment (merlin-northern) [inline tests/tests/test_management_config.py:76]: I would love to see two more cases:
1. from the existing configuration set on key to empty value, so:
* first we PUT:
```json
configuration = {
            "key": "value",
            "another-key": "another-value",
        }
```
* we check that it is there with GET
* we put empty value:
```json
configuration = {
            "key": "value",
            "another-key": "",
        }
```
* we verify with GET that it is returned as empty.

2. what happens when you PUT empty config:
```json
configuration = {
        }
```

**Review by merlin-northern — CHANGES_REQUESTED** [review]
I believe we need a couple of more scenarios.

File: tests/tests/test_management_config.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    assert r.status == 204
+
+
+class TestManagementConfig:
+    def test_config_device(self, device_id):
+        user_id = str(uuid.uuid4())
+        client = management_api_with_params(user_id=user_id)
+        #
+        # get the configuration (empty)
+        r = client.get_key_value_pair_store(device_id)
```
Comment (alfrunes) [inline tests/tests/test_management_config.py:44]: (Unrelated) Can we change the api spec operation ID to something more succinct (e.g `Get Device Configuration`)? This sounds a bit strange to me.

**Review by alfrunes — COMMENTED** [review]
The tests looks great, but on top of what @merlin-northern said I would also like to request a test case where setting configuration values with type different than string returns an error.

File: tests/tests/test_management_config.py (hunk 2 above)
Comment (tranchitella) [reply tests/tests/test_management_config.py]: @alfrunes done

File: tests/tests/test_management_config.py (hunk 1 above)
Comment (tranchitella) [reply tests/tests/test_management_config.py:77]: @merlin-northern I implemented (1), (4) and (5); (3) is already in place, we always check if reported is empty, in all the tests. (2) is missing from your list :)

**Review by merlin-northern — APPROVED** [review]
lgtm! thank you

**Comment by tranchitella** [comment]
I'm stopping the QA pipeline, it has no value in this context.

**Comment by tranchitella** [comment]
@alfrunes I'm waiting for your feedback before merging.

**Review by alfrunes — APPROVED** [review]
Good enough for me :+1:

**Pull request merged**